### DISCUSSION TITLE
File upload when the form data is converted to an Object

### DISCUSSION THREAD

**Discussion by JossLadanyi** [pocketbase/pocketbase#2032] [Q&A] [answered]
Hi,

I can upload a file when the form containing the file is retrived by formData()

const formData = await request.formData();
book = await locals.pb.collection('books').create(formData);

but is it possible to upload it if the form is converted to an Object?
I would like to manipulate some of the other properties beforehand and the way I found to do it was to convert the form to an Object:

const formData = await request.formData();
const formObject = Object.fromEntries(formData);
book = await locals.pb.collection('books').create(formObject);

create works and all other properties seem to get set but the file is not uploaded. Should I do some transformation on the file property in formObject before calling create?

Thank you for your time.

Kind regards,

Joseph

**Comment by ganigeorgiev** [comment, marked as answer]
> but is it possible to upload it if the form is converted to an Object?

No. File upload is supported only over `multipart/form-data` requests and you'll have to submit a [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) body.

This was recently discussed in https://github.com/pocketbase/pocketbase/discussions/2004#discussioncomment-5239827 and eventually we may add an auto "Object -> FormData" conversion to the SDK in the future, but for now you have to construct the `FormData` instance on your own.


But in your case, since you already have a `FormData` object, you don't necessarily need to convert it to a new plain object if you want to modify it.
You can extract specific value(s) with `formData.get("yourField")`, apply your transformation(s) and then update the old value with `formData.set("yourField", newValue)`.

**Reply by JossLadanyi** [reply]
Thank you so much for your prompt answer, as always!
I should manipulate the formData object directly, then.
Thanks again!!!
